feat: add special platform event

Decodetalkers opened the pull request:
There are many different events in different platform, so they should not included in any of the events, like touch, window or etc. This maybe will make the production platform plugin more flexible

The core team is busy and does not have time to mentor nor babysit new contributors. If a member of the core team thinks that reviewing and understanding your work will take more time and effort than writing it from scratch by themselves, your contribution will be dismissed. It is your responsibility to communicate and figure out how to reduce the likelihood of this!

Read the contributing guidelines for more details: https://github.com/iced-rs/iced/blob/master/CONTRIBUTING.md

Because we are trying to make iced on different wayland platforms, so ..

hecrj reviewed (commented):
What's the specific use case here?

Decodetalkers committed:
feat: add custom key for iced Event

There are many different events in different platform, so they should
not included in any of the events, like touch, window or etc. This maybe
will make the production platform plugin more flexible

Decodetalkers committed:
chore: make emulator able for all custom type

Decodetalkers committed:
chore: remove default genertic in Runtime

Decodetalkers committed:
chore: bring custom to other events

Decodetalkers committed:
chore: set the default genertic type of EventStream to ()

Decodetalkers committed:
chore: move Custom to future event

Decodetalkers committed:
chore: add function to listen on platform unique events

Decodetalkers committed:
chore: do fmt

Decodetalkers committed:
chore!: move PlatformSpectic to iced_winit

Decodetalkers committed:
fix: backend do not pass Custom and fix examples

Decodetalkers committed:
chore: ci fix

Decodetalkers committed:
fix: remove useless generic
